### PR TITLE
Create an extensive doc on config customizations in this chart

### PR DESCRIPTION
**Pull request by petewall**

### CODE REVIEW

File: charts/k8s-monitoring/docs/Customizations.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+* `metrics.extraMetricsRelabelingRules` - Same as above, but the rules are applied to metrics from all metric sources.
+* `metrics.<source>.allowList` - Sets a list of metrics that will be kept, dropping any metrics that don't match.
+* `extraServices.prometheus.externalLabels` - A key-value set that defines labels and values to be set for all metrics
+  being sent. It sets the `external_labels` section of
+  [prometheus.remote_write](https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.remote_write/#arguments)
+  component.
+
+### Delivery customizations
+
+These fields adjust how metrics are actually delivered to the metrics service. Tyipcally, these will be changed only to
```
Comment (BeverlyJaneJ) [inline charts/k8s-monitoring/docs/Customizations.md:61]:
```suggestion
These fields adjust how metrics are actually delivered to the metrics service. Typically, these will be changed only to
```

File: charts/k8s-monitoring/docs/Customizations.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+tune the performance of metrics uploads.
+
+* `extraServices.prometheus.processors` - These processors will control the batch and memory size of metrics to bundle
```
Comment (BeverlyJaneJ) [inline charts/k8s-monitoring/docs/Customizations.md:64]:
```suggestion
* `extraServices.prometheus.processors` - These processors control the batch and memory size of metrics to bundle
```

File: charts/k8s-monitoring/docs/Customizations.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  before sending them to the service. Only applies to metrics services using the OTLP or OTLPHTTP protocols.
+* `extraServices.prometheus.wal` - This controls the behavior of the Write Ahead Log that's used in the remote_write
+  component.
+
+## Logs
+
+### Discovery customizations
+
+Customizing the discovery of logs will affect which Pods to gather logs from. The default is all pods in all namespaces,
+but this can easily be adjusted.
```
Comment (BeverlyJaneJ) [inline charts/k8s-monitoring/docs/Customizations.md:74]:
```suggestion
but you can easily adjust this setting.
```

File: charts/k8s-monitoring/docs/Customizations.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+* `logs.pod_logs.namespaces` - Only gather logs from pods in the given list of namespaces.
```
Comment (BeverlyJaneJ) [inline charts/k8s-monitoring/docs/Customizations.md:76]:
```suggestion
* `logs.pod_logs.namespaces` - Only gather logs from Pods in the given list of namespaces.
```

File: charts/k8s-monitoring/docs/Customizations.md [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+* `logs.pod_logs.extraRelabelingRules` - These rules are used to filter from all pods on the cluster to the specific set
```
Comment (BeverlyJaneJ) [inline charts/k8s-monitoring/docs/Customizations.md:77]:
```suggestion
* `logs.pod_logs.extraRelabelingRules` - These rules are used to filter from all Pods on the Cluster to the specific set
```

File: charts/k8s-monitoring/docs/Customizations.md [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+  that will be used for gathering logs. They're used to populate the rules section of a
+  [discovery.relabel](https://grafana.com/docs/agent/latest/flow/reference/components/discovery.relabel/) component.
+
+### Gathering customizations
+
+Customizing this field controls how pod logs are actually gathered from the cluster.
```
Comment (BeverlyJaneJ) [inline charts/k8s-monitoring/docs/Customizations.md:83]:
```suggestion
This field controls how Pod logs are gathered from the Cluster.
```

File: charts/k8s-monitoring/docs/Customizations.md [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+
+* `logs.pod_logs.gatherMethod` - The default method, "volumes", means that the Grafana Agent for Logs will gather logs
```
Comment (BeverlyJaneJ) [inline charts/k8s-monitoring/docs/Customizations.md:85]:
```suggestion
* `logs.pod_logs.gatherMethod` - The default method, "volumes", means that the Grafana Agent for logs will gather logs
```

File: charts/k8s-monitoring/docs/Customizations.md [hunk 8]
```diff
@@ -0,0 +1,114 @@
+# Customizing the Configuration
+
+By default, this chart provides a very capable configuration for Kubernetes Cluster infrastructure monitoring. However
```
Comment (BeverlyJaneJ) [inline charts/k8s-monitoring/docs/Customizations.md:3]:
```suggestion
By default, this chart provides a very capable configuration for Kubernetes Cluster infrastructure monitoring. However,
```

File: charts/k8s-monitoring/docs/Customizations.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+
+Metrics are typically gathered using Prometheus-style scraping.
+
+### Discovery customizations
+
+These fields in the Helm chart's values affect how metric targets are discovered. Typically, you'll make changes here if
+the Grafana Agent is not detecting the target.
+
+* `metrics.<source>.extraRelabelingRules` - These rules are used to filter from all nodes, services, endpoints, or pods on the
+  cluster to a specific one that has the metrics. They're used to populate the rules section of a
```
Comment (BeverlyJaneJ) [inline charts/k8s-monitoring/docs/Customizations.md:26]:
```suggestion
  Cluster to a specific one that has the metrics. They're used to populate the rules section of a
```

File: charts/k8s-monitoring/docs/Customizations.md [hunk 10]
```diff
@@ -0,0 +1,114 @@
...
+4. Delivery - Where should the data be sent for storage?
+
+## Metrics
+
+Metrics are typically gathered using Prometheus-style scraping.
+
+### Discovery customizations
+
+These fields in the Helm chart's values affect how metric targets are discovered. Typically, you'll make changes here if
+the Grafana Agent is not detecting the target.
```
Comment (BeverlyJaneJ) [inline charts/k8s-monitoring/docs/Customizations.md:23]:
```suggestion
These fields affect how metric targets are discovered. Typically, you'll make changes to these values if the Grafana Agent is not detecting the target.
```

**Review by BeverlyJaneJ — CHANGES_REQUESTED** [review]

**Comment by petewall** [comment]
Merged your changes. Thank you so much, @BeverlyJaneJ !

File: charts/k8s-monitoring/docs/Customizations.md [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+Each telemetry data source (e.g. metrics, logs, events, traces) typically goes through the same process:
+
+1. Discovery - How does the collector find the data source?
+2. Scraping - How should the data get gathered?
+3. Processing - What work needs to be done to the data?
+4. Delivery - Where should the data be sent for storage?
+
+## Metrics
+
+Metrics are typically gathered using Prometheus-style scraping.
```
Comment (BeverlyJaneJ) [inline charts/k8s-monitoring/docs/Customizations.md:16]:
```suggestion
Metrics are usually gathered using Prometheus-style scraping.
```

File: charts/k8s-monitoring/docs/Customizations.md [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+
+### Discovery customizations
+
+These fields affect how metric targets are discovered. Typically, you'll make changes to these values if the Grafana Agent is not detecting the target.
```
Comment (BeverlyJaneJ) [inline charts/k8s-monitoring/docs/Customizations.md:20]:
```suggestion
These fields affect how metric targets are discovered. If the Grafana Agent is not detecting the target, make changes to these values.
```

Comment (BeverlyJaneJ) [reply charts/k8s-monitoring/docs/Customizations.md]: In this section, do we need to give some examples of what issues users are trying to solve, such as reducing metrics? For eample, why would I use relabeling for one source as opposed to all?

File: charts/k8s-monitoring/docs/Customizations.md [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+These fields affect how metric targets are scraped. Typically, you'll make changes here if 
+you want to change something about how often to scrape or which target to scrape.
+
+* `metrics.<source>.scrapeInterval` - How often to scrape metrics from a target.
+* `metrics.scrapeInterval` - Same as above, but changes the global default.
+* `metrics.<source>.service.isTLS` - Determines whether the Agent use HTTPS and TLS to scrape the target instead of HTTP.
+
+### Processing customizations
+
+By default, we already use an "allow list" to filter to a specific set of metrics, meaning the list drops metrics that are not useful for monitoring Kubernetes Clusters. The following fields allow for more customizations of the metrics after they have been scraped, but before being sent to the external metric service for storage. Typical changes that you might want to do here include setting, modifying, or dropping metric labels.
```
[the remark this answers is not in the document]
Comment (BeverlyJaneJ) [reply charts/k8s-monitoring/docs/Customizations.md:41]: Again, wondering if we need to say why a user would want to do this.

File: charts/k8s-monitoring/docs/Customizations.md [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+* `metrics.<source>.allowList` - Sets a list of metrics that will be kept, dropping any metrics that don't match.
+* `extraServices.prometheus.externalLabels` - A key-value set that defines labels and values to be set for all metrics
+  being sent. It sets the `external_labels` section of
+  [prometheus.remote_write](https://grafana.com/docs/agent/latest/flow/reference/components/prometheus.remote_write/#arguments)
+  component.
+
+### Delivery customizations
+
+These fields adjust how metrics are delivered to the metrics service. Typically, change these only to
+tune the performance of metrics uploads.
```
[the remark this answers is not in the document]
Comment (BeverlyJaneJ) [reply charts/k8s-monitoring/docs/Customizations.md:56]: What's the problem I'm solving?

File: charts/k8s-monitoring/docs/Customizations.md [hunk 15]
```diff
@@ -0,0 +1,107 @@
...
+
+* `extraServices.prometheus.processors` - These processors control the batch and memory size of metrics to bundle
+  before sending them to the service. Only applies to metrics services using the OTLP or OTLPHTTP protocols.
+* `extraServices.prometheus.wal` - Controls the behavior of the Write Ahead Log that's used in the remote_write
+  component.
+
+## Logs
+
+### Discovery customizations
```
Comment (BeverlyJaneJ) [inline charts/k8s-monitoring/docs/Customizations.md:65]:
```suggestion
### Customize which logs to gather
```

Comment (BeverlyJaneJ) [reply charts/k8s-monitoring/docs/Customizations.md:65]: Probably need an intro sentence here - why would I want to customize log discovery?

File: charts/k8s-monitoring/docs/Customizations.md [hunk 16]
```diff
@@ -0,0 +1,107 @@
...
+
+When you customize the discovery of logs, it will affect which Pods to gather logs from. The default is all Pods in all namespaces,
+but you can easily adjust this setting.
+
+* `logs.pod_logs.namespaces` - Only gathers logs from Pods in the given list of namespaces.
+* `logs.pod_logs.extraRelabelingRules` - Rules that filter from all Pods on the Cluster to the specific set
+  that will be used for gathering logs. They're used to populate the rules section of a
+  [discovery.relabel](https://grafana.com/docs/agent/latest/flow/reference/components/discovery.relabel/) component.
+
+### Gathering customizations
```
[the remark this answers is not in the document]
Comment (BeverlyJaneJ) [reply charts/k8s-monitoring/docs/Customizations.md:75]: Let's talk to make this section more understandable

File: charts/k8s-monitoring/docs/Customizations.md [hunk 17]
```diff
@@ -0,0 +1,107 @@
...
+
+This field controls how Pod logs are gathered from the Cluster.
+
+* `logs.pod_logs.gatherMethod` - The default method, "volumes", means that the Grafana Agent for logs will gather logs
+  by mounting hostPath volumes to the Pod log location on each Kubernetes Cluster Node. The other method, "api", means
+  that the Agent will gather logs by streaming them from the Kubernetes API Server.
+
+### Processing customizations
```
Comment (BeverlyJaneJ) [inline charts/k8s-monitoring/docs/Customizations.md:83]:
```suggestion
### Customize how logs are processed
```

File: charts/k8s-monitoring/docs/Customizations.md [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+
+* `logs.pod_logs.extraStageBlocks` - Processing logs is done in stages. This field allows for additional stages to
+  be set. Stages set here will be used to populate a
+  [loki.process](https://grafana.com/docs/agent/latest/flow/reference/components/loki.process/) component.
+
+## Events
+
```
[the remark this answers is not in the document]
Comment (BeverlyJaneJ) [reply charts/k8s-monitoring/docs/Customizations.md:90]: Let's talk about this section as well.

**Review by BeverlyJaneJ — APPROVED** [review]

**Pull request merged**